YOLayout/YOLayout.m

```diff
@@ -65,7 +65,17 @@ - (CGSize)sizeThatFits:(CGSize)size {
 }
 
 - (CGRect)setFrame:(CGRect)frame view:(id)view sizeToFit:(BOOL)sizeToFit {
-  return [self setFrame:frame view:view options:(sizeToFit ? YOLayoutOptionsSizeToFit : 0)];
+
+  YOLayoutOptions options = 0;
+  // If UILabel and 1 line, we need to fix size to fit options to constrain width,
+  // since UILabel will return a size > the width we specify.
+  if (sizeToFit && [view isKindOfClass:UILabel.class] && ((UILabel *)view).numberOfLines == 1) {
```

Comment:
Yeah good points... So we could constrain width by default (for all things in sizeToFit:YES) except if the width is 0 (or throw an error or tell the person to use options:, because they are doing something "different".

I think for the CGFLOAT_MAX case for width, thats already required when say you are sizing text vertically and you don't want to have a (scroll) limit... So I'm ok with that one.

Also this is for sizeToFit:YES which is a convenience method for the very common case, so I think its ok to have it constrain width as long as that the most common thing.

We could also name this method, instead of setFrame:view:, sizeToFitInFrame:view: or setFrameToFit:view:. Since you would never do sizeToFit:NO?